Add new fields in smb2-create-request event

mauropalumbo75 opened the pull request:
I added the following additional fields to the smb2-create-request event:

access_mask : count;
share_access : count;
contexts : vector of SMB2::CreateContextValue;

Baseline for btest was also updated.

sethhall commented on scripts/base/init-bare.bro:
```diff
@@ -3358,6 +3358,20 @@ export {
+        	## Possible values are: ExtA, SecD, DHnQ, etc.
+		name        : string;
+	    	## The data payload of the type specified in the structure name.
+		data        : string;
```
Should this data be parsed internally?  I vaguely recall running into this and avoiding it at the time because I felt like I would have needed to parse the data based on the name field in order to make it most useful to users.

jsiwek commented on src/analyzer/protocol/smb/smb2-com-create.pac:
```diff
@@ -17,6 +36,18 @@ refine connection SMB_Conn += {
+
+		        	for ( auto i = 0u; i < ${val.contexts}->size(); ++i )
+	                		cv->Assign(i, BuildSMB2CreateContextVal(${val.contexts[i]}));
+                		requestinfo->Assign(5, cv);
```
The `contexts` field of `SMB2::CreateRequest` isn't `&optional`, so should unconditionally assign it a `VectorVal`.  Whether that value gets populated or remains empty is the logic that should be conditional.

jsiwek commented:
The `scripts.base.protocols.smb.smb1` unit test fails with the patch.

mauropalumbo75 commented:
I'll fix the above issues asap

jsiwek commented on testing/btest/Baseline/scripts.base.protocols.smb.smb1/smb_files.log:
```diff
@@ -3,10 +3,9 @@
-1403194574.150293	CHhAvVGS1DHFjwGM9	192.168.1.78	55770	192.168.1.53	445	-	SMB::FILE_OPEN	-	Test\\2009-12 Payroll.xlsx	25940	-	1403148950.000000	1403193623.046524	1403148950.000000	1403193632.973276
-#close	2016-08-05-15-25-54
+1403194573.484358	CHhAvVGS1DHFjwGM9	192.168.1.78	55770	192.168.1.53	445	-	SMB::FILE_OPEN	-	<share_root>	0	-	1403193605.830789	1403193605.830789	1403193211.405450	1403193605.830789
+1403194573.484701	CHhAvVGS1DHFjwGM9	192.168.1.78	55770	192.168.1.53	445	-	SMB::FILE_OPEN	-	Test	0	-	1403193632.973275	1403193632.973275	1403193604.628965	1403193632.973275
```
Just checking: the line that included `Test\\2009-12 Payroll.xlsx` I think was legit, but now missing likely due to the previous change to the parser grammar (the .pac file).  i.e. wondering if just updating the baseline is sufficient here, rather than fixing the underlying reason for the difference.

mauropalumbo75 commented:
You are absolutely right and I'll have to look into it with more attention.
Not sure though when I'll be able to do it, please be patient

On Mon, 29 Apr 2019, 19:47 Jon Siwek, <notifications@github.com> wrote:

> *@jsiwek* requested changes on this pull request.
> ------------------------------
>
> In testing/btest/Baseline/scripts.base.protocols.smb.smb1/smb_files.log
> <https://github.com/zeek/zeek/pull/338#discussion_r279467916>:
>
> >  #fields	ts	uid	id.orig_h	id.orig_p	id.resp_h	id.resp_p	fuid	action	path	name	size	prev_name	times.modified	times.accessed	times.created	times.changed
>  #types	time	string	addr	port	addr	port	string	enum	string	string	count	string	time	time	time	time
> -1403194573.483536	CHhAvVGS1DHFjwGM9	192.168.1.78	55770	192.168.1.53	445	-	SMB::FILE_OPEN	-	<share_root>	0	-	1403193605.830790	1403193605.830790	1403193211.405449	1403193605.830790
> -1403194573.484701	CHhAvVGS1DHFjwGM9	192.168.1.78	55770	192.168.1.53	445	-	SMB::FILE_OPEN	-	Test	0	-	1403193632.973276	1403193632.973276	1403193604.628965	1403193632.973276
> -1403194574.150293	CHhAvVGS1DHFjwGM9	192.168.1.78	55770	192.168.1.53	445	-	SMB::FILE_OPEN	-	Test\\2009-12 Payroll.xlsx	25940	-	1403148950.000000	1403193623.046524	1403148950.000000	1403193632.973276
> -#close	2016-08-05-15-25-54
> +1403194573.484358	CHhAvVGS1DHFjwGM9	192.168.1.78	55770	192.168.1.53	445	-	SMB::FILE_OPEN	-	<share_root>	0	-	1403193605.830789	1403193605.830789	1403193211.405450	1403193605.830789
> +1403194573.484701	CHhAvVGS1DHFjwGM9	192.168.1.78	55770	192.168.1.53	445	-	SMB::FILE_OPEN	-	Test	0	-	1403193632.973275	1403193632.973275	1403193604.628965	1403193632.973275
>
> Just checking: the line that included Test\\2009-12 Payroll.xlsx I think
> was legit, but now missing likely due to the previous change to the parser
> grammar (the .pac file). i.e. wondering if just updating the baseline is
> sufficient here, rather than fixing the underlying reason for the
> difference.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/zeek/zeek/pull/338#pullrequestreview-231811820>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEXYCJ62X6EAH6US5UNZYJTPS4YEZANCNFSM4HGCWCWQ>
> .
>

0xxon commented:
Hi,

I just wanted to check if you still plan working on this/if we should keep the merge request open.

This is separate from your second merge-request, right?

mauropalumbo75 commented:
Hii,
   I am still planning to work on this, but I can't really tell you when
sorry. I am trying to cope with a backlog of work to do and I can work on
this only in my spare time.
And yes this is separate from the other pull request on SMB though they
both come from a common request.

Best wishes,
Mauro

On Tue, May 21, 2019 at 6:14 PM Johanna <notifications@github.com> wrote:

> Hi,
>
> I just wanted to check if you still plan working on this/if we should keep
> the merge request open.
>
> This is separate from your second merge-request, right?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/zeek/zeek/pull/338?email_source=notifications&email_token=AEXYCJ5YFDFLS5AKS4NC74DPWQNXPA5CNFSM4HGCWCW2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODV4NKWA#issuecomment-494458200>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEXYCJZ3W3UQVZG2TYDARG3PWQNXPANCNFSM4HGCWCWQ>
> .
>

0xxon commented:
Hi,

as this has been open without any progress for a few months, I am going to close this. If you address the issues that have been raised, please re-open it and we will work on getting this merged.
